### PR TITLE
Prepare for per-workspace quota

### PR DESCRIPTION
**Pull request by ncdc**

### CODE REVIEW

File: pkg/controller/resourcequota/resource_quota_controller_patch.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+				newResources[k] = v
+			}
+		} else {
+			// short circuit in non-discovery error cases or if discovery returned zero resources
+			return
+		}
+	}
+
+	// Decide whether discovery has reported a change.
+	if reflect.DeepEqual(oldResources, newResources) {
```
Comment (stevekuznetsov) [inline pkg/controller/resourcequota/resource_quota_controller_patch.go:52]: is this in a critical path? probably couple orders of magnitude faster to loop

**Review by stevekuznetsov — COMMENTED** [review]
Seems reasonable, but if we can break up the middle commit into a copy and a diff, that would be best for rebasing

File: pkg/controller/resourcequota/resource_quota_controller_patch.go (hunk 1 above)
Comment (ncdc) [reply pkg/controller/resourcequota/resource_quota_controller_patch.go:52]: It's from upstream, but if we can improve performance, 👍.

Comment (stevekuznetsov) [reply pkg/controller/resourcequota/resource_quota_controller_patch.go:52]:
```go
updated := len(oldResources) != len(newResources)
if len(oldResources) == len(newResources) {
    for old := range oldResources {
        if _, has := newResources[old]; !has {
            updated = true
        }
    }
}
```

was what I meant

File: pkg/controller/resourcequota/resource_quota_controller_patch.go [hunk 2]
```diff
@@ -1,86 +1,92 @@
 /*
- Copyright 2022 The KCP Authors.
+Copyright 2022 The KCP Authors.
 
- Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (stevekuznetsov) [inline pkg/controller/resourcequota/resource_quota_controller_patch.go]: Likely not the diff you want ;)

Comment (ncdc) [reply pkg/controller/resourcequota/resource_quota_controller_patch.go]: Thanks, had some whitespace issues. Will fix.

Comment (ncdc) [reply pkg/controller/resourcequota/resource_quota_controller_patch.go]: Fixed

**Review by stevekuznetsov — APPROVED** [review]

**Comment by ncdc** [comment]
Replaced by #84

**Pull request closed without merging**